android/expoview/src/main/java/host/exp/exponent/RNObject.kt

```diff
@@ -0,0 +1,314 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+package host.exp.exponent
+
+import host.exp.exponent.analytics.EXL
+import host.exp.expoview.BuildConfig
+import java.lang.reflect.Constructor
+import java.lang.reflect.Field
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+
+// TODO: add type checking in DEBUG
+class RNObject {
```

Comment:
It seems like we could make this proxy class less error-prone with reified generics later

Reply:
Yep, agreed.